designs/2021-suppression-support/README.md

```diff
@@ -0,0 +1,511 @@
+  "results": [
+    {
+      "level": "error",
+      "message": {
+        "text": "'a' is not defined."
+      },
+      "locations": [
+        {
+          "physicalLocation": {
+            "artifactLocation": {
+              "uri": "file:///C:/Users/yiweiding/Documents/projects/test/1.js",
+              "index": 0
+            },
+            "region": {
+              "startLine": 1,
+              "startColumn": 1
+            }
+          }
+        }
+      ],
+      "suppressions": [
+        {
+          "kind": "inSource",
+          "justification": "Sample justification message."
+        }
+      ],
+      "ruleId": "no-undef",
+      "ruleIndex": 0
+    }
+  ]
+}
+```
+
+```json
+JSON format
+[
+  {
+    "filePath":"C:\\Users\\yiweiding\\Documents\\projects\\test\\1.js",
+    "messages":[
+      {
+        "ruleId":"no-undef",
+        "severity":2,
+        "message":"'a' is not defined.",
+        "line":1,
+        "column":1,
+        "nodeType":"Identifier",
+        "suppressions":[
+          {
+            "kind":"inSource",
+            "justification":"Sample justification message."
+          }
+        ],
+        "messageId":"undef",
+        "endLine":1,
+        "endColumn":2
+      }
+    ],
+    "errorCount":0,
+    "warningCount":0,
+    "fixableErrorCount":0,
+    "fixableWarningCount":0,
+    "usedDeprecatedRules":[]
+  }
+]
+```
+
+## Motivation
+
+### Problem
+
+ESLint is a famous tool for identifying and reporting on patterns found in ECMAScript/JavaScript code. When codes violate one or more defined rules, ESLint will report warnings or errors:
+
+```json
+"messages": [
+  {
+    "ruleId":"no-undef",
+    "severity":2,
+    "message":"'b' is not defined.",
+    "line":2,
+    "column":1,
+    "nodeType":"Identifier",
+    "messageId":"undef",
+    "endLine":2,
+    "endColumn":2
+  }
+]
+```
+
+![Violation](violation.png)
+
+In our scenarios, when violations are suppressed, Security Development Lifecycle (SDL) tools, such as ESLint, are expected to export suppression information, including its kind and justification. We could take advantage of this info to generate corresponding signals for auditing purposes.
+
+```json
+"messages": [
+  {
+    "ruleId":"no-undef",
+    "severity":2,
+    "message":"'a' is not defined.",
+    "line":2,
+    "column":1,
+    "nodeType":"Identifier",
+    "messageId":"undef",
+    "endLine":2,
+    "endColumn":2,
+    "suppressions": [
+      {
+        "kind":"inSource",
+        "justification":"Sample justification message."
+      }
+    ]
+  }
+]
+```
+
+### Suppression in ESLint
+
+There are 2 ways to suppress (or disable in ESLint) warnings/errors (violations), according to [disabling-rules](https://github.com/eslint/eslint/blob/master/docs/user-guide/configuring/rules.md#disabling-rules) and [comments in GitHub](https://github.com/eslint/eslint/issues/14784#issuecomment-878522942):
+
+1. **inSource**: use directive comments inside of a file
```

Comment:
Can we call this “directive” to better fit with ESLint terminology? You will be able to convert to “inSource” in your formatter?